### ISSUE TITLE
Remove trial's setUpClass and tearDownClass features

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4175]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4175|
|Type|task|
|Created|2009-12-17 00:47:25Z|
|Branch|https://github.com/twisted/twisted/tree/remove-class-fixtures-4175|

These two hooks have been deprecated ([#2303](https://github.com/twisted/twisted/issues/2303)) since 8.2.  There's significant complexity associated with continuing to support them, and in particular making disttrial ([#1784](https://github.com/twisted/twisted/issues/1784)) work with them would be more effort than it's worth.  We should remove the feature now.

<details><summary>Searchable metadata</summary>

```
trac-id__4175 4175
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_remove_class_fixtures_4175 branches/remove-class-fixtures-4175
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1261010845000000 1261010845000000
changetime__1364139159000000 1364139159000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [27798]) Branching to 'remove-class-fixtures-4175'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Done.  Lots of nice crufty code deleted.  Very enjoyable.  Hope I got it all.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-class-fixtures-4175).

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml set owner to @exarkun|
|-|-|

Wuu! I've been looking forward to this day for a long time. I envy you for being the one to delete these.

Some possible changes:
 1. The NEWS addition is good, but I personally would mention that a lot of crufty code is deleted, and perhaps mention the other bugs that are obsoleted by this change. Your discretion.
 1. AIUI, we require a blank line after docstrings. At least, PEP 8 does. Please add such a blank line before `test_hashability` in `test_testcase`.
 1. Consider splitting `test_hashability` into two separate tests.

Only the second one is mandatory, and that only if I've interpreted the coding standards correctly. I'm happy for you to make these changes at your discretion and land the branch without another review roundtrip.

Some other comments:
 * Thanks for adding the docstring to the `test_results` test for skipping.
 * Now that we've got rid of tearDownClass, perhaps we can remove the distinction between `TestCase._cleanUp` and `TestCase._classCleanUp`. For another ticket though.
 * Similarly, `upDownError` should probably be destroyed. Will that let us delete `BrokenTestCaseWarning`, I wonder.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [27804]) Split part of test_hashability off into test_equality

refs [#4175](https://github.com/twisted/twisted/issues/4175)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Thanks for the review!

I lean towards leaving the news entry as it is now because it is describing the user-facing changing adequately.  I feel like discussion of implementation issues (despite being pretty great, deleting ~600 lines of pretty gnarly code) don't really belong.  Since the other two tickets were closed as _wontfix_, it also feels out of place to mention them.

I followed up on the PEP 8 thing, and we discussed it on IRC, leading to the revelation that PEP 8 actually says something different, stupid, and wrong about (modern) emacs' `fill-paragraph`.  So, leaving the docstring as-is.

I did split `test_hashability` in two, though.  Good call.

Getting rid of `upDownError` sounds pretty cool too.  As long as it doesn't get in my way for [#1784](https://github.com/twisted/twisted/issues/1784), I'll leave that removal for you. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [27805]) Merge remove-class-fixtures-4175

Author: exarkun
Reviewer: jml
Fixes: [#4175](https://github.com/twisted/twisted/issues/4175)

Remove trial's `setUpClass` and `tearDownClass` features (deprecated since 8.2), and
all of the (~600 lines of) crufty code previously supporing them.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
